core/schema/schema.go

```diff
@@ -0,0 +1,51 @@
+package schema
+
+import "context"
+
+type Metadata struct {
+	Authority     string
+	Format        string
+	Compatibility string
+}
+
+type SchemaInfo struct {
+	ID       string `json:"id"`
+	Version  int32  `json:"version"`
+	Location string `json:"location"`
+}
+
+type SchemaFile struct {
+	ID     string
+	Types  []string
+	Fields []string
+	Data   []byte
+}
```

Comment:
What about just `Info` and `File`? So calling this would be less stuttered `schema.Info` and `schema.File`

Reply:
Yes, plan to clean this in another PR. For now, keeping some names as they were before.